### ISSUE TITLE
Training Kerple Parameters with Constraints

### ISSUE THREAD

**Issue by szhengac** [NVIDIA/NeMo#7039]
Hi,

I see that NeMo implements [Kerple](https://arxiv.org/pdf/2205.09921.pdf). According to the paper, there are some constraints on the valid values for its learnable parameters. For example, [`self.kerple_p`](https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/nlp/modules/common/megatron/position_embedding/kerple_relative_position_embedding.py#L69) needs to fall into `(0, 2]`. Though there is no details on how the authors post such constraint during the training, I would expect we need to use projected gradient descent type methods to learn these parameters. But I don't see such related code in NeMo. Do I miss anything? Thanks.
